### PR TITLE
KaizenTask1061_add_rps

### PR DESCRIPTION
**Pull request by tkpratardan**
Issue #1061 

I found different versions of the RPS formula, so going through the paper that gave this formula ([Epstein](https://journals.ametsoc.org/view/journals/apme/8/6/1520-0450_1969_008_0985_assfpf_2_0_co_2.xml?tab_body=pdf)) and [Constantinou](https://www.researchgate.net/publication/227378917_Solving_the_Problem_of_Inadequate_Scoring_Rules_for_Assessing_Probabilistic_Football_Forecast_Models), I found the approach mentioned in the latter made more sense than the one mentioned in [here ](https://journals.sagepub.com/doi/pdf/10.1177/1471082X18817650)

### CODE REVIEW

File: research_amp/soccer_prediction/sota_baseline_glm.py [hunk 1]
```diff
@@ -228,3 +228,42 @@ def dixon_coles_adjustment(home_goals, away_goals, mu_home, mu_away, rho):
...
     )
+
+
+def calculate_rps(df: pd.DataFrame) -> float:
+    """
+    Calculate the Rank Probability Score (RPS) for three-way outcome predictions in a DataFrame.
+
+    :param df: DataFrame containing the columns:
+                       'prob_home_win', 'prob_draw', 'prob_away_win',
+                       'actual_outcome'.
```
Comment (gpsaggese) [inline research_amp/soccer_prediction/sota_baseline_glm.py:239]: Can we pass the column names? Ok to use default vals.

File: research_amp/soccer_prediction/sota_baseline_glm.py [hunk 2]
```diff
@@ -228,3 +228,42 @@ def dixon_coles_adjustment(home_goals, away_goals, mu_home, mu_away, rho):
...
+            actuals = np.array([0, 0])
+        # Calculate cumulative actual outcomes.
+        cum_actuals = actuals.cumsum()
+        # Calculate RPS for the current match.
+        rps = (np.sum((cum_probs - cum_actuals) ** 2)) / 2
+        # Add the RPS of the current match to the total RPS sum.
+        rps_sum += rps
+    # Average RPS over all matches.
+    rps_avg = rps_sum / M
+    _LOG.info("RPS value for the model: %.4f", rps_avg)
```
Comment (gpsaggese) [inline research_amp/soccer_prediction/sota_baseline_glm.py:266]: IMO it's a debug

File: research_amp/soccer_prediction/sota_baseline_glm.py [hunk 3]
```diff
@@ -228,3 +228,42 @@ def dixon_coles_adjustment(home_goals, away_goals, mu_home, mu_away, rho):
 rasoprut.evaluate_model_predictions(
         final_df_with_dixon["actual_outcome"], final_df_with_dixon["predicted_outcome"]
     )
+
+
+def calculate_rps(df: pd.DataFrame) -> float:
```
Comment (gpsaggese) [inline research_amp/soccer_prediction/sota_baseline_glm.py:233]: can we add a few unit tests?

Comment (gpsaggese) [inline research_amp/soccer_prediction/sota_baseline_glm.py:233]: Let's move it to lib.

**Comment by sonaalKant** [comment]
Can you connect the PR with the issue so that it is easier to navigate.

**Review by sonaalKant — APPROVED** [review]

**Pull request merged**